pkg/aaq-controller/aaq-gate-controller/aaq-gate-controller_test.go

```diff
@@ -44,14 +44,23 @@ var _ = Describe("Test aaq-gate-controller", func() {
 			aaqjcInterfaceMock := client.NewMockAAQJobQueueConfigInterface(ctrl)
 			aaqjcInterfaceMock.EXPECT().Create(context.Background(), aaqjqc, metav1.CreateOptions{}).Return(aaqjqc, nil).Times(1)
 			cli.EXPECT().AAQJobQueueConfigs(testNs).Return(aaqjcInterfaceMock).Times(1)
-
+			namespaceLister := testsutils.FakeNamespaceLister{Namespaces: map[string]*corev1.Namespace{testNs: {ObjectMeta: metav1.ObjectMeta{Name: testNs}}}}
 			recorder := record.NewFakeRecorder(100)
-			qc := setupAAQGateController(cli, nil, nil, aaqjcInformer, recorder)
+			qc := setupAAQGateController(cli, nil, nil, aaqjcInformer, namespaceLister, recorder)
 			err, es := qc.execute(testNs)
 			Expect(err).ToNot(HaveOccurred())
 			Expect(es).To(Equal(Forget))
 		})
 
+		It("the key namespace doesn't exist should forget key", func() {
```

Comment:
maybe reword this to something like "should forget the key if its namespace doesn't exist" ?